pom.xml

```diff
@@ -1034,6 +1017,36 @@
                 </exclusions>
             </dependency>
 
+            <dependency>
+                <groupId>com.google.cloud.bigdataoss</groupId>
+                <artifactId>util-hadoop</artifactId>
+                <version>hadoop2-${dep.gcs.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>com.google.cloud.bigdataoss</groupId>
+                <artifactId>gcsio</artifactId>
+                <version>${dep.gcs.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpclient</artifactId>
+                <version>4.5.11</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpcore</artifactId>
+                <version>4.4.13</version>
+            </dependency>
+
+            <dependency>
+                <groupId>com.google.code.gson</groupId>
+                <artifactId>gson</artifactId>
+                <version>2.8.6</version>
+            </dependency>
+
```

Comment:
All these dependencies should be redundant if you are using shaded GCS connector.

Reply:
They are there due to enforcer errors. Without those dependencies amazon-aws-sdk and a few other artifacts complain about upper bounds.

Reply:
If the versions for gson, httpclient and httpcore dependencies are not explicitly defined in dependency management, maven-enforcer fails with the following upper bound dependency violations:

```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.apache.httpcomponents:httpclient:4.5.9 paths to dependency are:
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.amazonaws:aws-java-sdk-core:1.11.749
    +-org.apache.httpcomponents:httpclient:4.5.9
and
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcs-connector:hadoop2-2.1.3
    +-com.google.oauth-client:google-oauth-client:1.30.6
      +-com.google.http-client:google-http-client:1.34.2
        +-org.apache.httpcomponents:httpclient:4.5.11
, 
Require upper bound dependencies error for org.apache.httpcomponents:httpcore:4.4.11 paths to dependency are:
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.amazonaws:aws-java-sdk-core:1.11.749
    +-org.apache.httpcomponents:httpclient:4.5.9
      +-org.apache.httpcomponents:httpcore:4.4.11
and
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcs-connector:hadoop2-2.1.3
    +-com.google.oauth-client:google-oauth-client:1.30.6
      +-com.google.http-client:google-http-client:1.34.2
        +-org.apache.httpcomponents:httpcore:4.4.13
, 
Require upper bound dependencies error for com.google.code.gson:gson:2.2.4 paths to dependency are:
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcsio:2.1.3
    +-com.google.code.gson:gson:2.2.4
and
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcsio:2.1.3
    +-com.google.protobuf:protobuf-java-util:3.11.4
      +-com.google.code.gson:gson:2.8.6
and
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcsio:2.1.3
    +-io.grpc:grpc-alts:1.29.0
      +-io.grpc:grpc-core:1.29.0
        +-com.google.code.gson:gson:2.8.6
```

Reply:
If the gcsio version is not explicitly defined in dependencyManagement, maven-enforcer fails with the following upper bound dependency violations:

```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for io.grpc:grpc-context:1.22.1 paths to dependency are:
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcs-connector:hadoop2-2.1.3
    +-com.google.oauth-client:google-oauth-client:1.30.6
      +-com.google.http-client:google-http-client:1.34.2
        +-io.opencensus:opencensus-api:0.24.0
          +-io.grpc:grpc-context:1.22.1
and
+-io.prestosql:presto-hive:334-SNAPSHOT
  +-com.google.cloud.bigdataoss:gcs-connector:hadoop2-2.1.3
    +-com.google.cloud.bigdataoss:gcsio:2.1.3
      +-io.grpc:grpc-auth:1.29.0
        +-io.grpc:grpc-api:1.29.0
          +-io.grpc:grpc-context:1.29.0
```

Reply:
If util-hadoop is not explicitly defined then presto-hive fails with an used but undeclared dependency error.

Reply:
I think it's better to exclude these dependencies from the shaded gcs-connector dependency jar instead of adding them as dependencies to the project - they are anyway already shaded in the GCS connector jar and are not used at runtime.

Reply:
Sounds great, I'll try that. Thanks!

Reply:
@medb, @electrum , I only use the shaded jar but it seems unwieldy to use the shaded path for the static auth constants, ex. `com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.util.HadoopCredentialConfiguration.ACCESS_TOKEN_PROVIDER_IMPL_SUFFIX`. What do you think?

Reply:
Agreed, shading should only be for internal details. These are part of the public API so it’s strange to use that shaded names. Let’s stick with the non-shaded artifact for now.

Reply:
`HadoopCredentialConfiguration` class was accidentally excluded from public API in recent refactoring. It will be exposed in public API in next release after https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/390 will be merged.

Reply:
Sounds good. Should I go back to the unshaded artifact or do we want to wait until the next gcs-connector release? I can update to use the unshaded one quickly if we want to go that route.

Reply:
The fix we needed was introduced in gcs-connector 2.1.1, for use with performance cache enabled (disabled by default).

Reply:
I think it depends on next Presto release date, if it scheduled after new GCS connector release in 2 weeks or so, then it makes sense to wait and update straight to shaded GCS connector 2.1.4. But if next Presto release is earlier, then we can go ahead with unshaded artifacts for now, I think.

Reply:
I'll update the pr with non-shaded artifacts and save this one, can always go back to it - we'll see which one happens first:)

Reply:
Thanks @medb!

Reply:
GCS connector 2.1.4 was just released, so you can update this PR to use it now.

Reply:
Great!